src/main/java/ch/njol/skript/expressions/ExprNamed.java

```diff
@@ -59,27 +61,51 @@ public class ExprNamed extends PropertyExpression<Object, Object> {
 	@SuppressWarnings("null")
 	private Expression<String> name;
 	
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		setExpr(exprs[0]);
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (exprs[0].getReturnType().equals(ItemStack.class)) {
+			setExpr(exprs[0].getConvertedExpression(ItemType.class));
```

Comment:
Is this needed? Can you add a comment explaining why it is if so

Reply:
I have no idea what it is, what it does or why it's there, I didn't write it.

Reply:
Existing code had casting to ensure the type was not an ItemStack. For some reason in the past an ItemType could have been an ItemStack. This enforces that and removes the need for casting in runtime. This was discussed on the skriptlang Discord a long time ago to do it this way.